### PR TITLE
PR:  Backup and Restore Test #6 

### PR DESCRIPTION
**Pull request by aot29**
# MaRDI Pull Request

**Goal**: 
Backup script should be able to restore the database. For this, the backup service now shares mounts with wikibase. 

**Changes**:
Changed docker-compose.yml (and the other compose files too). Please check at least the docker-compose file.

**Instructions for PR review**:
(these are marked by the person creating the PR and should be checked by the reviewer) 
- [x] Conceptual Review (Logic etc...) 
- [x] Code Review (Review your implementation) 
- [ ] Checkout (Test changes locally)

### CODE REVIEW

File: docker-compose-dev.yml [hunk 1]
```diff
@@ -95,13 +95,24 @@ services:
...
       - ${BACKUP_DIR:-./backup}:/data
+      # only for running tests
+      - ./test:/test
     environment:
       DB_HOST: mysql.svc # internal docker hostname (alias) of the database service
       DB_NAME: ${DB_NAME}
       DB_USER: ${DB_USER}
       DB_PASS: ${DB_PASS}
       BACKUP_SCHEDULE: ${BACKUP_SCHEDULE:-15 5 * * *}
+      KEEP_DAYS: ${KEEP_DAYS:-100}
```
Comment (Hyper-Node) [inline docker-compose-dev.yml:115]: An option to toggle backups on and off would be great, especially if running everything locally.

File: test/test_backup.sh [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+    /app/restore.sh -t xml
+    
+    # Check that wiki is running and accessible
+    response=$(_get_wiki_http_response_code)
+    if [[ ! $response == '200' ]]; then
+        printf " - Test restore XML FAILED: Could not restore wiki at ${WIKI_MAIN_URL}.\n"
+        exit 1
+    fi    
+    printf ' - Test restore XML OK: Wiki was restored from XML dump.\n'
+}
```
Comment (Hyper-Node) [inline test/test_backup.sh:111]: Suggestion for other test, restore backuped files and also check the accumulated file-size.

**Review by Hyper-Node — APPROVED** [review]
Seems ok, just some suggestions.

**Review by physikerwelt — APPROVED** [review]

File: docker-compose-dev.yml (hunk 1 above)
Comment (physikerwelt) [reply docker-compose-dev.yml:115]: you could set that in a docker-composer.override file

File: test/test_backup.sh (hunk 2 above)
Comment (physikerwelt) [reply test/test_backup.sh:111]: future work. Maybe we should focus on bugfixes first.

File: docker-compose-ci.yml [hunk 3]
```diff
@@ -28,3 +28,8 @@ services:
     container_name: mardi-selenium
     volumes:
       - ./test:/test
+
```
[the remark this answers is not in the document]
Comment (aot29) [reply docker-compose-ci.yml:31]: Checksums is a good idea.

Increemental backups I don't know, let's discuss it.

[the remark this answers is not in the document]
Comment (aot29) [reply docker-compose-ci.yml:31]: Tools: 
* Duplicacy (check license)
* other open source tools?

**Pull request merged**